components/src/core/Drawer/DrawerNavItem.tsx

```diff
@@ -0,0 +1,232 @@
+
+    const hasAction = Boolean(onItemSelect || onClick || expandHandler);
+    const onClickAction = (): void => {
+        if (onItemSelect) {
+            onItemSelect();
+        }
+        if (onClick) {
+            onClick();
+        }
+        if (expandHandler) {
+            expandHandler();
+        }
+    };
+
+    const rightComponent = navItem.rightComponent || (chevron && !items) ? <ChevronRight /> : undefined;
+
+    function getActionComponent(): JSX.Element {
+        if (!items) {
+            return null;
+        }
+        return (
+            <div
+                onClick={(e): void => {
+                    if (e) {
+                        expandHandler();
+                    }
+                }}
+                className={clsx(defaultClasses.expandIcon, classes.expandIcon, {
+                    [defaultClasses.expanded]: !collapseIcon && expanded,
+                })}
+            >
+                {collapseIcon && expanded ? collapseIcon : expandIcon}
+            </div>
+        );
+    }
+
+    const actionComponent = getActionComponent();
+
+    // 0 indents for top level nav items
+    // 0, 2, 4, ... for secondary level and beyond
+    const paddingLeft = theme.spacing(depth ? ((depth - 1) * 4) : 0);
+
+    const active = activeItem === itemID;
+
+    return (
+        <div
+            style={{ position: 'relative' }}
+            className={clsx(classes.listItemContainer, active && defaultClasses.listItemNoHover)}
+        >
+            {active && (
+                <div
+                    className={clsx(defaultClasses.active, classes.active, {
+                        [defaultClasses.square]: activeItemBackgroundShape === 'square',
+                    })}
+                    style={{ backgroundColor: activeItemBackgroundColor }}
+                />
+            )}
+            <InfoListItem
+                dense
+                title={itemTitle}
+                subtitle={itemSubtitle}
+                divider={divider ? 'full' : undefined}
+                statusColor={statusColor}
+                fontColor={active ? activeItemFontColor : itemFontColor}
+                icon={icon}
+                iconColor={active ? activeItemIconColor : itemIconColor}
+                rightComponent={
+                    (actionComponent || rightComponent) && (
+                        <div style={{ display: 'flex', alignItems: 'center', color: itemIconColor }}>
+                            {rightComponent}
+                            {actionComponent}
+                        </div>
+                    )
+                }
+                backgroundColor={'transparent'}
+                onClick={hasAction ? onClickAction : undefined}
+                style={{ paddingLeft }}
+                hidePadding={hidePadding}
+                ripple={ripple}
+                classes={depth > 0 ? { title: clsx(defaultClasses.nestedTitle, classes.nestedTitle) } : {}}
+                {...InfoListItemProps}
```

Comment:
we should merge our classes with their classes...otherwise if they want to put a class on something other than the nested items, it will blow away our nested class/styles.

Reply:
I added a fix to combine the classes for nested Items.  If they pass in a class to the InfoListItem title, it wont be applied to the nested items.